service/authorization.go

```diff
@@ -0,0 +1,109 @@
+//
+// This file is part of Clouditor Community Edition.
+
+package service
+
+import (
+	"context"
+
+	"clouditor.io/clouditor/api/orchestrator"
+	"golang.org/x/exp/slices"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/golang-jwt/jwt/v4"
+	grpc_auth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+)
+
+type RequestType int
+
+const (
+	AccessCreate RequestType = iota
+	AccessRead
+	AccessUpdate
+	AccessDelete
+)
+
+// AuthorizationStrategy is an interface that implements a function which whether the current cloud service request can
+// be fulfilled using the current access strategy.
+type AuthorizationStrategy interface {
+	CheckAccess(ctx context.Context, typ RequestType, req orchestrator.CloudServiceRequest) bool
+	AllowedCloudServices(ctx context.Context) (all bool, IDs []string)
+}
+
+// AuthorizationStrategyJWT is an AuthorizationStrategy that expects a list of cloud service IDs to be in a specific JWT
+// claim key.
+type AuthorizationStrategyJWT struct {
+	Key string
+}
+
+// CheckAccess checks whether the current request can be fulfilled using the current access strategy.
+func (a *AuthorizationStrategyJWT) CheckAccess(ctx context.Context, typ RequestType, req orchestrator.CloudServiceRequest) bool {
+	var list []string
+
+	return slices.Contains(list, req.GetCloudServiceId())
+}
+
+// AllowedCloudServices retrieves a list of allowed cloud service IDs according to the current access strategy.
+func (a *AuthorizationStrategyJWT) AllowedCloudServices(ctx context.Context) (all bool, list []string) {
+	var err error
+	var token string
+	var claims jwt.MapClaims
+
+	// Retrieve the raw token from the context
+	token, err = grpc_auth.AuthFromMD(ctx, "bearer")
+	if err != nil {
+		return false, nil
+	}
+
+	// We need to re-parse the already validiated claim to get a specific key from the claims map.
+	parser := jwt.NewParser()
+	_, _, err = parser.ParseUnverified(token, &claims)
+	if err != nil {
+		return false, nil
+	}
+
+	for _, item := range claims[a.Key].([]interface{}) {
+		list = append(list, item.(string))
+	}
+
+	return false, list
+}
+
+// AuthorizationStrategyJWT is an AuthorizationStrategy that allows all requests.
+type AuthorizationStrategyAllowAll struct{}
+
+// CheckAccess checks whether the current request can be fulfilled using the current access strategy.
+func (a *AuthorizationStrategyAllowAll) CheckAccess(ctx context.Context, typ RequestType, req orchestrator.CloudServiceRequest) bool {
+	return true
+}
+
+// AllowedCloudServices retrieves a list of allowed cloud service IDs according to the current access strategy.
+func (a *AuthorizationStrategyAllowAll) AllowedCloudServices(ctx context.Context) (bool, []string) {
+	return true, nil
+}
+
+// ErrPermissionDenied represents an error, where permission to fulfill the request is denied.
+var ErrPermissionDenied = status.Errorf(codes.PermissionDenied, "access denied")
```

Comment:
It is a lil bit lonely there. Maybe
- put it to the top
- add it to api/errors.go (but I think we do not follow this "convention"  here like in CAM)